### PR TITLE
vlog.go: Fix wrongly maintain the offset of file in memory (#640)

### PR DESCRIPTION
**Pull request by july2993**
### What problem does this PR solve? 
cherry-pick https://github.com/pingcap/tidb-binlog/pull/640

### What is changed and how it works?
cherry-pick https://github.com/pingcap/tidb-binlog/pull/640

### Check List 

Tests 

 - Unit test
Code changes

Side effects

 Related changes

- Need to be included in the release note

### CODE REVIEW

**Comment by GregoryIan** [comment]
LGTM

**Comment by july2993** [comment]
/run-all-tests tidb=release-2.1 tikv=release-2.1 pd=release-2.1

File: pump/storage/vlog.go [hunk 1]
```diff
@@ -298,6 +298,8 @@ func (vlog *valueLog) write(reqs []*request) error {
 
 	toDisk := func() error {
 		n, err := curFile.fd.Write(vlog.buf.Bytes())
+		atomic.AddInt64(&vlog.writableLogOffset, int64(n))
+
```
Comment (suzaku) [inline pump/storage/vlog.go]: Should we also make sure that `fsync` is called when `n > 0`?

**Pull request merged**